### PR TITLE
ARROW-1744: [WIP] Add plasma tensorflow op

### PR DESCRIPTION
**Pull request by pcmoritz**
Before this can be merged, it needs some profiling and comparison to other ways of loading data in TensorFlow.

### CODE REVIEW

File: cpp/src/plasma/tf/test.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import pyarrow as pa
+import pyarrow.plasma as plasma
+import tensorflow as tf
+
+import time
+
+zero_out_module = tf.load_op_library('./plasma_op.so')
+
+client = plasma.connect("/tmp/plasma", "", 64)
```
Comment (holdenk) [inline cpp/src/plasma/tf/test.py]: Minor comment, but maybe do a mktemp type thing here instead? I know some folks run tests in parallel on the same machine.

File: cpp/src/plasma/tf/plasma_op.cc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    std::cout << "shape is" << result->shape()[0] << " , " << result->shape()[1]
+              << std::endl;
+
+    // Create an output tensor
+    TensorShape shape(result->shape());
+    Tensor* output_tensor = NULL;
+    OP_REQUIRES_OK(context, context->allocate_output(0, shape, &output_tensor));
+    auto output_flat = output_tensor->flat<float>();
+
+    // Set all but the first element of the output tensor to 0.
```
Comment (holdenk) [inline cpp/src/plasma/tf/plasma_op.cc]: Is this comment perhaps about an older version of the code?

**Review by holdenk — COMMENTED** [review]
So I'm still new to this project, and my C++ is a little rusty but I had a few questions :)

**Comment by xhochy** [comment]
@pcmoritz Any progress on this topic here?

**Comment by pcmoritz** [comment]
replaced by https://github.com/apache/arrow/pull/2046

**Pull request closed without merging**